### PR TITLE
[BTAT-9821]: Implemented feature switch, connector, http parser, service, model

### PR DESCRIPTION
**Pull request by BackRowOla**

### CODE REVIEW

**Comment by BackRowOla** [comment]
App config base PR: #6380

File: test/connectors/httpParsers/PenaltiesHttpParserSpec.scala [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    "the HTTP response status is 5xx" should {
+      val body: JsObject = Json.obj(
+        "code" -> "GATEWAY_TIMEOUT",
+        "message" -> "GATEWAY_TIMEOUT"
+      )
+
+      val httpResponse = HttpResponse(Status.GATEWAY_TIMEOUT, body.toString())
+      val expected = Left(ServerSideError(Status.GATEWAY_TIMEOUT.toString, httpResponse.body))
+      val result = DirectDebitStatusReads.read("", "", httpResponse)
```
Comment (william-hatzar) [inline test/connectors/httpParsers/PenaltiesHttpParserSpec.scala:112]: Should this not be PenaltiesReads?

File: test/connectors/httpParsers/PenaltiesHttpParserSpec.scala [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    "the HTTP response status isn't handled" should {
+      val body: JsObject = Json.obj(
+        "code" -> "Conflict",
+        "message" -> "CONFLCIT"
+      )
+
+      val httpResponse = HttpResponse(Status.CONFLICT, body.toString())
+      val expected = Left(UnexpectedStatusError("409", httpResponse.body))
+      val result = DirectDebitStatusReads.read("", "", httpResponse)
```
Comment (william-hatzar) [inline test/connectors/httpParsers/PenaltiesHttpParserSpec.scala:127]: Same here?

File: app/connectors/httpParsers/PenaltiesHttpParser.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+object PenaltiesHttpParser extends ResponseHttpParsers {
+
+  implicit object PenaltiesReads extends HttpReads[HttpGetResult[PenaltiesSummary]] {
+    override def read(method: String, url: String, response: HttpResponse): HttpGetResult[PenaltiesSummary] = {
+      response.status match {
+        case OK => Right(response.json.as[PenaltiesSummary])
+        case NOT_FOUND => Right(PenaltiesSummary.empty)
+        case BAD_REQUEST => handleBadRequest(response.json)
+        case status  if status >= 500 & status < 600 => Left(ServerSideError(response.status.toString, response.body))
```
Comment (william-hatzar) [inline app/connectors/httpParsers/PenaltiesHttpParser.scala:33]: This is not getting picked up by the coverage report but I believe it's to do with a small error in the parserSpec tests, which is mentioned below.

File: app/connectors/httpParsers/PenaltiesHttpParser.scala [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        case _ => Left(UnexpectedStatusError(response.status.toString, response.body))
```
Comment (william-hatzar) [inline app/connectors/httpParsers/PenaltiesHttpParser.scala:34]: This is also not getting picked up and I believe that it is to do with the same reason.

File: test/connectors/httpParsers/PenaltiesHttpParserSpec.scala (hunk 1 above)
Comment (BackRowOla) [reply test/connectors/httpParsers/PenaltiesHttpParserSpec.scala:112]: Yes it should, making the change and will push up

File: test/connectors/httpParsers/PenaltiesHttpParserSpec.scala (hunk 2 above)
Comment (BackRowOla) [reply test/connectors/httpParsers/PenaltiesHttpParserSpec.scala:127]: same as above

File: app/connectors/httpParsers/PenaltiesHttpParser.scala (hunk 3 above)
Comment (BackRowOla) [reply app/connectors/httpParsers/PenaltiesHttpParser.scala:33]: The tests are now being covered. It was as you mentioned with the parserSpec tests

File: app/connectors/httpParsers/PenaltiesHttpParser.scala (hunk 4 above)
Comment (BackRowOla) [reply app/connectors/httpParsers/PenaltiesHttpParser.scala:34]: Same as above

**Review by william-hatzar — APPROVED** [review]

**Pull request merged**